control-plane/agents/src/bin/core/node/service.rs

```diff
@@ -361,6 +361,11 @@ impl Service {
 
     /// Apply a drain label to the specified node. The reconciler will perform the drain.
     async fn drain(&self, id: NodeId, label: String) -> Result<Node, SvcError> {
+        // Don't allow draining if HA_ENABLED is false. If it is undefined we treat it as true.
+        if std::env::var("HA_ENABLED").unwrap_or("true".to_string()) == "false" {
```

Comment:
This won't be used internally, only externally.
Let's add an option to the cli args, and pass them to the registry. 
We can make the cli option: `--disable-ha` (you can also add DISABLE_HA env variable as you have here)

Then we can make the check here for service calls and also within the node drain itself just in case it ever gets called some other way.

Reply:
Done. Btw @tiagolobocastro I did not disable republish as core-agent ha tests use republish without HA components.

Reply:
Although, since we decided the default value would be true, it would work. Should I disable republish what do you think?

Reply:
> Done. Btw @tiagolobocastro I did not disable republish as core-agent ha tests use republish without HA components.

That's fine I think, because we wouldn't  disable ha on the core agent

Reply:
> Although, since we decided the default value would be true, it would work. Should I disable republish what do you think?

Yes exactly, I think we should disable it.

Reply:
Done.